src/protocols/ethereum/composables/ethMaxAmount.ts

```diff
@@ -0,0 +1,37 @@
+import { Ref, computed } from 'vue';
+import BigNumber from 'bignumber.js';
+
+import { BigNumberPublic } from '@/types';
+import {
+  MaxAmountOptions,
+  useBalances,
+} from '@/composables';
+import { ETH_CONTRACT_ID } from '../config';
+
+interface EthMaxAmountOptions extends MaxAmountOptions {
+  fee: Ref<BigNumberPublic>;
+}
+
+/**
+ * Composable that allows to use real max amount of selected token
+ * considering the fee that needs to be paid.
+ */
+export function useEthMaxAmount({ formModel, fee }: EthMaxAmountOptions) {
+  const { balance } = useBalances();
+
+  const isEthCoin = computed(() => formModel.value?.selectedAsset?.contractId === ETH_CONTRACT_ID);
+  // TODO converted balance is probably wrong, as it represents the balance of the token in ETH
+  const selectedTokenBalance = computed(
+    () => new BigNumber(formModel.value?.selectedAsset?.convertedBalance || 0),
+  );
+
+  const max = computed(() => {
+    if (balance.value && isEthCoin.value) {
+      const _max = balance.value.minus(fee.value);
+      return (_max.isPositive() ? _max : 0).toString();
```

Comment:
I think using underscores to avoid name collisions is rather bad practice. I suggest to come up with something more useful :)
Maybe `maxBalance`, `availableBalance` or something else?

Reply:
This code was inspired by existing `maxAmount` composable. But I agree with your suggestion